s3/src/main/java/com/yahoo/ycsb/db/S3Client.java

```diff
@@ -0,0 +1,538 @@
+ * s3.endPoint=s3.amazonaws.com
+ * s3.region=us-east-1
+ * The parameter table is the name of the Bucket where to upload the files. 
+ * This must be created before to start the benchmark
+ * The size of the file to upload is determined by two parameters:
+ * - fieldcount this is the number of fields of a record in YCSB
+ * - fieldlength this is the size in bytes of a single field in the record
+ * together these two parameters define the size of the file to upload, 
+ * the size in bytes is given by the fieldlength multiplied by the fieldcount.
+ * The name of the file is determined by the parameter key. 
+ *This key is automatically generated by YCSB.
+ *
+ */
+public class S3Client extends DB {
+ 
+  private static AmazonS3Client s3Client;
+  private static String sse;
+  private static SSECustomerKey ssecKey;
+  private static final AtomicInteger INIT_COUNT = new AtomicInteger(0);
+  //private static int initCount = 0;
+
+  /**
+  * Cleanup any state for this storage.
+  * Called once per S3 instance; 
+  */
+  @Override
+  public void cleanup() throws DBException {
+    if (INIT_COUNT.decrementAndGet() == 0) {
+      try {
+        s3Client.shutdown();
+        System.out.println("The client is shutdown successfully");
+      } catch (Exception e){
+        System.err.println("Could not shutdown the S3Client: "+e.toString());
+        e.printStackTrace();
+      } finally {
+        if (s3Client != null){
+          s3Client = null;
+        }
+      }
+    }
+  }
+  /**
+  * Delete a file from S3 Storage.
+  * 
+  * @param bucket
+  *            The name of the bucket
+  * @param key
+  * The record key of the file to delete.
+  * @return Zero on success, a non-zero error code on error. See the
+  * {@link DB} class's description for a discussion of error codes.
+  */
+  @Override
+  public int delete(String bucket, String key) {
+    try {
+      s3Client.deleteObject(new DeleteObjectRequest(bucket, key));
+    } catch (Exception e){
+      System.err.println("Not possible to delete the key "+key);
+      e.printStackTrace();
+      return 1;
+    }
+    return 0;
+  }
+  /**
+  * Initialize any state for the storage.
+  * Called once per S3 instance; If the client is not null it is re-used.
+  */
+  @Override
+  public void init() throws DBException {
+    final int count = INIT_COUNT.incrementAndGet();
+    synchronized (S3Client.class){
+      Properties propsCL = getProperties();
+      int recordcount = Integer.parseInt(
+          propsCL.getProperty("recordcount"));
+      int operationcount = Integer.parseInt(
+          propsCL.getProperty("operationcount"));
+      int numberOfOperations = 0;
+      if (recordcount > 0){
+        if (recordcount > operationcount){
+          numberOfOperations = recordcount;
+        } else {
+          numberOfOperations = operationcount;
+        }
+      } else {
+        numberOfOperations = operationcount;
+      }
+      if (count <= numberOfOperations) {
```

Comment:
I think the parsing of the record and operation count can be removed and the ```if (count < numberofOperations)``` just replaced with ```if (count == 0)```.  e.g., do once.